src/components/BotAppearance/ColorPicker.tsx

```diff
@@ -0,0 +1,73 @@
+import {
+  Card,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Popover,
+  PopoverContent,
+  PopoverPortal,
+  PopoverTrigger,
+  Typography,
+} from '@mochi-ui/core'
+import { useEffect, useState } from 'react'
+import { SketchPicker } from 'react-color'
+import { Controller, useFormContext } from 'react-hook-form'
+import { type BotAppearance } from './BotAppearance'
+
+interface ColorData {
+  defaultColor: string
+}
+
+export const ColorPicker = (props: ColorData) => {
+  const { control, setValue } = useFormContext<BotAppearance>()
+  const [color, setColor] = useState(props.defaultColor)
+
+  useEffect(() => {
+    setColor(props.defaultColor)
+  }, [props.defaultColor])
+
+  return (
+    <Controller
+      name="accentColour"
+      control={control}
+      render={({ fieldState }) => (
+        <FormControl error={!!fieldState.error} hideHelperTextOnError>
+          <FormLabel>Color</FormLabel>
+          <Popover>
+            <PopoverTrigger asChild>
+              <Card asChild>
+                <div className="flex flex-row space-x-4 items-center cursor-pointer">
+                  <Card asChild>
+                    <div
+                      className="`w-4 h-4 rounded-lg"
+                      style={{ backgroundColor: color }}
+                    />
+                  </Card>
+                  <Typography level="p4" fontWeight="lg">
+                    {color}
+                  </Typography>
+                </div>
+              </Card>
+            </PopoverTrigger>
+            <PopoverPortal>
+              <PopoverContent asChild>
+                <div className="w-[300px] border-none bg-transparent shadow-none ">
```

Comment:
Can we keep the size?

![image](https://github.com/dwarvesf/chatbot-builder/assets/12707960/51ff5d42-1b4b-4d3c-9a1a-3db38ec33216)